testing/supporting_docs/ui_selectors_and_their_importance.md

```diff
@@ -0,0 +1,178 @@
+## The importance of UI Selectors
+There are various application engineering decisions that will affect the decision that can 
+be made when choosing an Appium client. For example, if engineering decides to add “preferred” 
+selectors such as ‘accessibility Id’s’ or ‘ID’s’ to the application that will be tested, the 
+main benifit will be a test suite that is more stable as disscussed in the next section. Also 
+a library with less selectors such as Webdriverio could be used to create the tests.  If 
+engineering decides not to take this path then, the test suite will likely be more flaky, and 
+the best client to use is the Java-client which has a wide range of selectors.
+
+
+### Prefered and Unadvisable Selectors
+
+Selectors such as ‘AccessibilityId’ or ‘ID’ are generally prefered when using 
+Appium testing. Due to the facts that they are relatively short, they generally 
+provide a indicator of what an element is used for, can be independently targeted 
+from parent elements, and will not break if the order of elements, or parents of 
+elements, on a UI page are modified. 
+
+Absolute xpaths tend to be more brittle an slower than other forms of selectors 
+and indeed Appium inspector will give warnings about the possible performance 
+issues of using an xpath when it is the only selector available.  The reasons for 
+such issues are that absolute xpaths contain class information or information from 
+parents that could change depending on the development of an application.  This 
+can be more prominent for hybrid apps or in browser testing where xpaths can become
+ much longer. 
+
+It is possible to improve the rigidness and performance of xpaths in test by using 
+relative xpaths that are shorter and more flexible.  For example, a standard xpath 
+selector from an Android application such as:
```

Comment:
@twnolan consistency should be persisted with the word 'android' across the whole document, e.g. starting with either lower or upper case.